src/conversations/Conversation.ts

```diff
@@ -4,40 +4,61 @@ import Client, {
   ListMessagesPaginatedOptions,
   SendOptions,
 } from '../Client'
-import Message from '../Message'
+import {
+  InvitationContext,
+  InvitationV1,
+  SealedInvitationHeaderV1,
+} from '../Invitation'
+import { MessageV1, MessageV2 } from '../Message'
+import { messageApi, xmtpEnvelope, fetcher } from '@xmtp/proto'
+import { encrypt, decrypt, SignedPublicKey, Signature } from '../crypto'
+import Ciphertext from '../crypto/Ciphertext'
+import { sha256 } from '../crypto/encryption'
+import { dateToNs, nsToDate } from '../utils'
+const { b64Decode } = fetcher
 
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
 
 /**
  * Conversation class allows you to view, stream, and send messages to/from a peer address
  */
-export default class Conversation {
-  peerAddress: string
+export class ConversationV1 {
+  private _peerAddress: string
+  sent: Date
+  context = null
   private client: Client
 
-  constructor(client: Client, address: string) {
-    this.peerAddress = address
+  constructor(client: Client, address: string, sent: Date) {
+    this._peerAddress = address
     this.client = client
+    this.sent = sent
+  }
+
+  async peerAddress(): Promise<string> {
```

Comment:
Making this async is going to be an annoying breaking change for developers. This field gets used a lot, often in synchronous code (like rendering a UI). I'd suggest we make this a concrete property and do the async work for the ConversationV2 before the class is initialized.